src/Services/FileManager.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Services;
+
+use Symfony\Component\HttpFoundation\File\Exception\FileException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\Filesystem\Filesystem;
+
+class FileManager
+{
+
+    /**
+     * @return array<string>
+     */
+    public function saveFile(string $fileName, UploadedFile $file, string $folder): array
+    {
+        $error = '';
+
+        $modifiedName = $this->getNewFileName($fileName);
+
+        if ($modifiedName === null) {
+            $modifiedName = 'partner';
+        }
+
+        $newFileName = $modifiedName . '-' . uniqid() . '.' . $file->guessExtension();
+
+        try {
+            $file->move(
+                $folder,
+                $newFileName
+            );
+        } catch (FileException $e) {
+            $error = 'Le fichier n\'a pas pu être ajouté.';
+        }
+
+        return ['error' => $error, 'fileName' => $newFileName];
+    }
+
+    public function deleteFile(string $fileName, string $folder): void
+    {
+        $filesystem = new Filesystem();
+        $path = $folder . $fileName;
+        $filesystem->remove($path);
+    }
+
+    public function getNewFileName(string $oldName, string $delimiter = '-'): ?string
+    {
+        $newName = iconv('UTF-8', 'ASCII//TRANSLIT', $oldName);
+
+        if ($newName != null) {
```

Comment:
You should have strict comparison by default

Reply:
Stan bother me again when I put !==, he say the same error message as before :   54     Parameter #3 $subject of function preg_replace expects array|string,  
         string|false given. 
So I let != and he's fine with it